app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt

```diff
@@ -768,27 +773,38 @@ class ProjectActivity : BaseActivity<ProjectViewModel.ViewModel>(), CancelPledge
     }
 
     private fun updateFragments(project: Project) {
-        val (rewardsFragment, backingFragment) = supportFragmentManager.findFragmentById(R.id.fragment_rewards) as RewardsFragment to
-                supportFragmentManager.findFragmentById(R.id.fragment_backing) as BackingFragment
-        when {
-            supportFragmentManager.backStackEntryCount == 0 -> when {
-                project.isBacking -> if (!rewardsFragment.isHidden) {
-                    supportFragmentManager.beginTransaction()
-                            .show(backingFragment)
-                            .hide(rewardsFragment)
-                            .commitNow()
-                }
-                else -> if (!backingFragment.isHidden) {
-                    supportFragmentManager.beginTransaction()
-                            .show(rewardsFragment)
-                            .hide(backingFragment)
-                            .commitNow()
+        try {
+            safeLet(rewardsFragment(), backingFragment()) { rewardsFragment, backingFragment ->
+                when {
```

Comment:
So the way I'm understanding `safeLet` is that this closure just won't execute if either of these values are `null`. Just wondering if it's necessary to define the new generic function for this purpose or just have that same `if (rewardsFragment() != null && backingFragment() != null)` just right here? Unless you feel it improves readability split out into a function.

Also - are there any threading concerns here? Do we need to be sure `block(p1, p2)` is executing on the main thread?

Reply:
yes that's what it does! but no it does not improve readability, i was just being fancy :3 

wrt threading; it executes in a `run` block so I think I'd be okay